akka-epl/src/main/scala/com/epl/akka_v1/HttpClient.scala

```diff
@@ -0,0 +1,85 @@
+package com.epl.akka_v1
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshalling.Marshal
+import akka.http.scaladsl.model.headers.BasicHttpCredentials
+import akka.http.scaladsl.model.{HttpMethods, HttpRequest, RequestEntity, headers}
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+import com.epl.akka.DocumentType
+
+import scala.concurrent.Future
+
+object HttpClient {
+
+  implicit val system = ActorSystem()
+  implicit val materializer = ActorMaterializer()
+  implicit val ec = system.dispatcher
+
+  final val config = system.settings.config
+
+  private val authorization = headers.Authorization(BasicHttpCredentials(config.getString("cloudant.username"), config.getString("cloudant.password")))
+
+
+  /**
+    * POST call to Cloudant database API
+    * @param jsonString
+    * @return
+    */
+  def post(jsonString: String): Future[String] = {
+    val responseEntity = for {
+      request <- Marshal(jsonString).to[RequestEntity]
+      response <- Http().singleRequest(HttpRequest(
+        method = HttpMethods.POST,
+        uri = config.getString("cloudant.post_url"),
+        headers = List(authorization),
+        entity = request))
+      entity <- Unmarshal(response.entity).to[ByteString]
+    } yield entity
+
+    responseEntity.mapTo[String]
+
+  }
+
+
+  /**
+    * GET call to Cloudant database API
+    * @param documentType
+    * @return
+    */
+  def get(documentType: DocumentType.Documenttype): Future[String] = {
+    val responseEntity = for {
+      response <- Http().singleRequest(HttpRequest(
+        method = HttpMethods.GET,
+        uri = getUrl(documentType),
+        headers = List(authorization)))
+      entity <- Unmarshal(response.entity).to[ByteString]
+    } yield entity
+
+    responseEntity.mapTo[String]
+
+  }
+
+
+  /**
+    * Get URI based on the type of document requested.
+    * @param documentType
+    * @return
+    */
+  def getUrl(documentType: DocumentType.Documenttype): String ={
+    var url: String = null
+    if(documentType.equals(DocumentType.TeamTable)){
```

Comment:
Perfect use case for Scala pattern matching, would make this
```
documentType match {
  case DocumentType.TeamTable => tablesUrl
  case DocumentType.Results => resultsUrl
  case DocumentType.Fixtures => fixturesUrl
}
```

additionally changing `DocumentType` into an ADT (algebraic data type), like so:
```
sealed trait DocumentType
case object TeamTable extends DocumentType
case object Results extends DocumentType
case object Fixtures extends DocumentType
```
would then give you a compiler error if you at some future point added another document type but forgot to update the pattern match expression to cover all possible values.